compiler/rustc_parse/src/parser/attr.rs

```diff
@@ -371,23 +362,7 @@ impl<'a> Parser<'a> {
     /// meta_item_inner : (meta_item | UNSUFFIXED_LIT) (',' meta_item_inner)? ;
     /// ```
     pub fn parse_meta_item(&mut self) -> PResult<'a, ast::MetaItem> {
-        let nt_meta = match &self.token.kind {
-            token::Interpolated(nt) => match &nt.0 {
-                token::NtMeta(e) => Some(e.clone()),
-                _ => None,
-            },
-            _ => None,
-        };
-
-        if let Some(item) = nt_meta {
-            return match item.meta(item.path.span) {
-                Some(meta) => {
-                    self.bump();
-                    Ok(meta)
-                }
-                None => self.unexpected(),
-            };
-        }
+        maybe_whole!(self, NtMeta, |attr| attr.meta(attr.path.span).expect("meta item"));
```

Comment:
The `.expect("meta item")` case is reachable.
`NtMeta` contains an `AttrItem`, which doesn't necessarily fit into the meta item syntax.

Something like this should trigger it.
```rust
macro_rules! mac {
    ($attr_item: meta) => {
        #[cfg($attr_item)]
        struct S;
    }
}

mac!(an(arbitrary token stream));
```

Reply:
Good catch. I have reverted the use of `maybe_whole` here because it won't work, though I have rewritten the existing code to more closely match the structure of `maybe_whole`.